src/main/java/com/google/api/codegen/util/ProtoParser.java

```diff
@@ -422,6 +422,57 @@ private String getResourceSetFullName(ResourceSet resource, ProtoFile file) {
     return String.format("%s.%s", resource.getSymbol(), getProtoPackage(file));
   }
 
+  public ImmutableMap<String, String> getFieldNamePatterns(Method method) {
+    // Only look two levels deep in the request object, so fields of fields of the request object.
+    return ImmutableMap.copyOf(getFieldNamePatterns(method.getInputMessage(), 2));
+  }
+
+  private Map<String, String> getFieldNamePatterns(MessageType messageType, int depth) {
```

Comment:
Can we add a method-level comment to describe what this method is trying to do?

Reply:
Done.